### PR TITLE
[flink] CDC Ingestion options change did not trigger schema update

### LINKED ISSUE #1951: [Bug] options change did not trigger schema update
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/incubator-paimon/issues) and found nothing similar.

### Paimon version

0.5

### Compute Engine

flink

### Minimal reproduce step

Synchronize the MySQL table to Paimon, stop the Flink program, modify the table-conf configuration to change 'tag.creation-period' from 'daily' to 'hourly', but the changes do not take effect.

### What doesn't meet your expectations?

Changing the parameter does not take effect, which is not as expected.

### Anything else?

_No response_

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by MonsterChenzhuo**
### Purpose

close #1951

The current PR has implemented the following: After canceling the single-table synchronization Flink job and modifying the settings, the changed or deleted table parameters take effect upon restart. For full database synchronization, in scenarios where new tables are added upon restart, the newly added tables can execute according to the new table configuration items.

### Tests

KafkaSchemaITCase#testOptionsChange()
MongoDBSyncTableActionITCase#testOptionsChange()
MySqlSyncTableActionITCase#testOptionsChange()
MySqlSyncDatabaseActionITCase#testNewlyAddedTablesOptionsChange()
MongoDBSyncDatabaseActionITCase#testNewlyAddedTablesOptionsChange()
KafkaSchemaITCase#testNewlyAddedTablesOptionsChange()

### CODE REVIEW

File: paimon-core/src/main/java/org/apache/paimon/tag/TagAutoCreation.java [hunk 1]
```diff
@@ -109,8 +109,13 @@ private TagAutoCreation(
         } else {
             Snapshot lastTag = tags.lastKey();
             this.nextSnapshot = lastTag.id() + 1;
-
-            LocalDateTime time = periodHandler.tagToTime(tags.get(lastTag));
+            LocalDateTime time;
+            String lastTagValue = tags.get(lastTag);
+            if (periodHandler instanceof DailyTagPeriodHandler) {
```
Comment (JingsongLi) [inline paimon-core/src/main/java/org/apache/paimon/tag/TagAutoCreation.java:114]: Is here a test for this?

**Comment by JingsongLi** [comment]
Hi @MonsterChenzhuo , I think the table config should be dynamic options, it should not be updated to the schema.

File: paimon-core/src/main/java/org/apache/paimon/tag/TagAutoCreation.java (hunk 1 above)
Comment (MonsterChenzhuo) [reply paimon-core/src/main/java/org/apache/paimon/tag/TagAutoCreation.java]: done.

**Comment by MonsterChenzhuo** [comment]
> Hi @MonsterChenzhuo , I think the table config should be dynamic options, it should not be updated to the schema.

done.

**Comment by MonsterChenzhuo** [comment]
@JingsongLi  PTAL,thanks.

**Review by JingsongLi — APPROVED** [review]
+1 Thanks @MonsterChenzhuo

**Pull request merged**